mlrun/utils/notifications/notification/mail.py

```diff
@@ -0,0 +1,119 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import json
+import typing
+from email.message import EmailMessage
+
+import aiosmtplib
+
+import mlrun.common.schemas
+import mlrun.lists
+import mlrun.utils.helpers
+
+from .base import NotificationBase
+
+DEFAULT_SMTP_PORT = 587
+
+
+class MailNotification(NotificationBase):
+    """
+    API/Client notification for sending run statuses as a mail message
+    """
+
+    required_params = [
+        "server_host",
+        "server_port",
+        "sender_address",
+        "username",
+        "password",
+        "email_addresses",
+    ]
+
+    @classmethod
+    def validate_params(cls, params):
+        for required_param in cls.required_params:
+            if required_param not in params:
+                raise ValueError(
+                    f"Parameter '{required_param}' is required for MailNotification"
+                )
+
+        if type(params["email_addresses"]) not in [str, list]:
+            raise ValueError(
+                "Parameter 'email_addresses' must be a string or a list of strings"
+            )
+
+    async def push(
+        self,
+        message: str,
+        severity: typing.Union[
+            mlrun.common.schemas.NotificationSeverity, str
+        ] = mlrun.common.schemas.NotificationSeverity.INFO,
+        runs: typing.Union[mlrun.lists.RunList, list] = None,
+        custom_html: typing.Optional[str] = None,
+        alert: mlrun.common.schemas.AlertConfig = None,
+        event_data: mlrun.common.schemas.Event = None,
+    ):
+        self.params.setdefault("subject", f"[{severity}] {message}")
+        self.params.setdefault("body", message)
+        await self._send_email(**self.params)
+
+    @staticmethod
+    async def _send_email(
+        email_addresses: str,
+        sender_address: str,
+        server_host: str,
+        server_port: int,
+        username: str,
+        password: str,
+        use_tls: bool,
+        subject: str = "MLRun Notification",
+        body: str = "",
+        **kwargs,
+    ):
+        # Create the email message
+        message = EmailMessage()
+        message["From"] = sender_address
+        message["To"] = email_addresses
+        message["Subject"] = subject
+        message.set_content(body)
+
+        # Send the email
+        await aiosmtplib.send(
+            message,
+            hostname=server_host,
+            port=server_port,
+            username=username,
+            password=password,
+            use_tls=use_tls,
```

Comment:
From the aiosmtplib docs:

TARTTLS connections
By default, if the server advertises STARTTLS support, aiosmtplib will upgrade the connection automatically. Setting use_tls=True for STARTTLS servers will typically result in a connection error.

So we probably need a way to let the user set start_tls as well.

Reply:
I think we should set start_tls to `False` anyway, because I don't think that the user needs to know all these implementation details.
If we pass `use_tls` as `True` it should be encrupted. 
Another question, what do you think about setting the `validate_certs` to the value of `use_tls` that I got from the user? is it correct or maybe I should add another params for `validate_certs` that by default will be `True`.
@adamdelman, @quaark , @liranbg , @theSaarco